packages/template-retail-react-app/app/hooks/use-basket.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useCustomerBaskets, useCustomerId, useProducts} from 'commerce-sdk-react-preview'
+
+/**
+ * This hook combine some commerce-react-sdk hooks to provide more derived data for Retail App baskets
+ * @param id - basket id to get the current used basket among baskets returned, use first basket in the array if not defined
+ * @param shouldFetchProductDetail - boolean to indicate if the baskets should fetch product details based on basket items
+ */
+export const useBasket = ({id = '', shouldFetchProductDetail = false} = {}) => {
+    const customerId = useCustomerId() || ''
+    const {
+        data: basketsData,
+        isLoading: isBasketsLoading,
+        error: isBasketsError,
+        ...restOfBaskets
+    } = useCustomerBaskets({customerId})
+    // if id is not defined, by default use the first basket in the list
+    const basket =
+        basketsData?.baskets?.find((basket) => basket.basketId === id) || basketsData?.baskets?.[0]
+    const productIds = '' ?? basket?.productItems?.map(({productId}) => productId).join(',')
+    const {data: products, isLoading: isProductsLoading} = useProducts(
+        {
+            ids: productIds,
+            allImages: true
+        },
+        {
+            enabled: shouldFetchProductDetail && !!productIds
+        }
+    )
+
+    return {
+        baskets: basketsData,
+        isBasketsLoading,
+        isBasketsError,
+        ...restOfBaskets,
+        // current picked basket
+        basket,
+        isLoading: isBasketsLoading && isProductsLoading,
```

Comment:
Should it be an OR operation instead?
```suggestion
        isLoading: isBasketsLoading || isProductsLoading,
```

Reply:
We would want to wait for both product details and baskets. On use case is in cart page, we need information from product detail to show cart items, so we have to wait until all data is available before showing them on the UI

Reply:
@alexvuong Ok, so it needs to be an `||` because this is `isLoading` and it's not like `isDone`.

Since we want to wait for all data to be available, isLoading will be false _only if_:
basketLoading=false || productLoading=false => false

Otherwise, when either one is true, isLoading is also true:
- basketLoading=false || productLoading=true => true
- basketLoading=true || productLoading=false => true

Reply:
Thanks for pointing it out, I've made the change as suggested